### PR TITLE
Adding units to stream_table

### LINKED ISSUE #322: Add units column to stream table in IFV
Add a column to the flowsheet viewer stream table that shows the units for a given row.

There is a possibility that the row may contain different units (ex. lb/ft and kg/m) so you may need to get the units of the first entry of the row and then convert the entire row to those units via `pyunits.convert(var[index], to_units=output_units)`

To get the units:
```
from pyomo.environ import units as pyunits

pyunits.get_units(var[index])
```

Note: `get_units` only works on an element of an indexed variable so you shouldn't use it on a variable container.

### PR DESCRIPTION
**Pull request by elbashandy**
Fixes #322 

## Summary/Motivation:
Adding units to the stream table view that shows up on the UI page. The units are generated randomly using the pyomo library to read the `derived_units` units from the FlowSheetBlock.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by dangunter** [comment]
This PR should be changed to be adding units, even if it is starting as mock units. A PR for adding mock units would never get merged.

File: idaes/ui/flowsheet.py [hunk 1]
```diff
@@ -449,6 +449,26 @@ def _construct_model_json(self):
             )
         )
 
+        if ('Units' in self._stream_table_df and
+            'Variable' in self._stream_table_df):
+            # Styling the units for display
+            def apply_html(value, style=''):
```
Comment (dangunter) [inline idaes/ui/flowsheet.py:455]: Wouldn't this make more sense in the JS client? Seems wrong to hardcode display styles in the server.

File: idaes/ui/fsvis/static/js/stream_table.js [hunk 2]
```diff
@@ -36,7 +36,17 @@ export class StreamTable {
             if (column_header !== "") {
```
Comment (dangunter) [inline idaes/ui/fsvis/static/js/stream_table.js:36]: could reduce indentation by doing a continue here instead

File: idaes/ui/fsvis/static/js/stream_table.js [hunk 3]
```diff
@@ -36,7 +36,17 @@ export class StreamTable {
...
                 // If the column_header is Variable then we don't want the column to be right-aligned and we want the column to be pinned to the left so when the user scrolls the column scrolls with them
                 if (column_header === "Variable") {
```
Comment (dangunter) [inline idaes/ui/fsvis/static/js/stream_table.js:38]: This is where you could take the plain text sent from the server and make it into appropriate HTML, e.g. if the server sent "VariableName,m/s^2" you could split on the comma and format.

File: idaes/ui/fsvis/static/css/main.css [hunk 4]
```diff
@@ -335,4 +335,13 @@ input:checked + .slider:before {
 
 .dropdown-item:active {
   background-color: white;
+}
+
```
Comment (dangunter) [inline idaes/ui/fsvis/static/css/main.css:339]: Please add some comments here saying what these affect and why. I would suggest more specific class names like "streamtable-variable"

**Review by dangunter — CHANGES_REQUESTED** [review]

File: idaes/core/util/tables.py [hunk 5]
```diff
@@ -245,10 +252,14 @@ def create_stream_table_dataframe(
             for i in disp_dict[k]:
                 if i is None:
```
Comment (dangunter) [inline idaes/core/util/tables.py:253]: from here to line 264 seems like duplicated code except for the difference in the key. why not do that up front and use the same logic, i.e. `units_key = k if is None else f"{k} {i}"` then use 'units_key' in place of 'k' and not require the `if i is None` decision

**Review by dangunter — APPROVED** [review]
LGTM

**Review by lbianchi-lbl — APPROVED** [review]
For testing, I called `idaes.ui.fsvis.visualize()` on the flowsheet created by `idaes.ui.tests.test_flowsheet.demo_flowsheet()`, and checked the stream table in the web UI. The units (and therefore this PR) LGTM!

image

**Pull request merged**